TicketUtil.py

```diff
@@ -119,6 +118,123 @@ def close_requests_session(self):
+        if self.auth != 'rest':
+            return super()
+        else:
+            try:
+                s = requests.Session()
+                r = s.get(self.auth_url, params=self.credentials, verify=False)
+                r.raise_for_status()
+                self.checkForBZError(r)
+                resp = r.json()
+                self.token = resp['token']
+                logging.debug("Create requests session: Status Code: {0}".format(r.status_code))
+                logging.info("Successfully authenticated to {0} with token: {1}".format(self.ticketing_tool, self.token))
+
+                return s
+            # We log an error if authentication was not successful, because rest of the HTTP requests will not succeed.
+
+            except requests.RequestException as e:
+                logging.error("Error authenticating to {0}. No valid credentials were provided.".format(self.auth_url))
+                logging.error(e.args[0])
+
+
+    def checkForBZError(self, bzRessponse):
+        if 'error' in bzRessponse.json() and bzRessponse.json()['error']:
+            raise Exception(bzRessponse.json()['message'] if 'message' in bzRessponse.json() else 'Unknown API error')
+
+
+    def create_ticket(self, params):
+        """
+        Tries to create the ticket through the ticketing tool's API.
+        Retrieves the ticket_id and creates the ticket_url.
+        :param params: The payload to send in the POST request.
+        :return:
+        """
+        # Attempt to create ticket.
+        try:
+            r = self.s.post(self.rest_url, params={'token': self.token}, json=params)
+
+            r.raise_for_status()
+            logging.debug("Create ticket: Status Code: {0}".format(r.status_code))
+
+            self.checkForBZError(r)
+
+            self.ticket_id = r.json()['id']
+            self.ticket_url = self.craft_ticket_url()
+            logging.info("Created ticket {0} - {1}".format(self.ticket_id, self.ticket_url))
+
+        # If ticket creation is not successful, log an error.
+        except requests.RequestException as e:
+            logging.error("Error creating ticket")
+            logging.error(e.args[0])
+
+    def craft_ticket_url(self):
+        ticket_url = None
+
+        if self.ticket_id:
+            ticket_url = "{0}/show_bug.cgi?id={1}".format(self.base_url, self.ticket_id)
+            print(ticket_url)
+
+        # If we do not receive a ticket ID in our initialization it indicates we'll be creating a ticket.
+        else:
+            ticket_url = None
+
+        return ticket_url
+
+    def add_comment(self, comment):
+        if not self.ticket_id: # Create the payload for our update.
+            logging.error("No ticket ID associated with ticket object. Set ticket ID with set_ticket_id(ticket_id)")
+            return
+
+        try:
+            params = {'comment': comment}
+
+            if self.token:
+                params.update({'token': self.token})
```

Comment:
While this approach is acceptable, for a simple assignment like this you would generally use something like params['token'] = self.token.
If you wanted to overwrite existing key/value pairs en masse or write key/values pairs where the value was another dictionary, then this approach would be better suited. See: https://docs.python.org/3.5/library/stdtypes.html#dict.update